pkg/cri/cri.go

```diff
@@ -87,13 +88,56 @@ func initCRIService(ic *plugin.InitContext) (interface{}, error) {
 		return nil, fmt.Errorf("failed to create containerd client: %w", err)
 	}
 
+	// guestPlatform in runtime struct for a runtime helps to support pulling of imager
+	// per runtime class. 
+	// guestPlatform is used to specify an alternate platform to use with platform matcher
+	// so that different versions of the same image can be pulled for different runtime handlers.
+	// guestPlatform.OS and guestPlatform.Architecture are compusory fields to be specified and
+	// if platform is windows, OSVersion needs to be specified too.
```

Comment:
I could see each platform owner, having a set of rules for how this would be used, yes.

i.e. armv7 vs armv8